src/v2/controllers/vreplicaset_controller/proof/helper_invariants/predicate.rs

```diff
@@ -157,30 +108,43 @@ pub open spec fn no_pending_create_or_delete_request_not_from_controller_on_pods
         }
     }
 }
-//
-// TODO: Prove this.
-//
-// Proving this for the VReplicaSet controller should be easy; we'd need to do a similar
-// proof for other state machines within the compound state machine.
-//
 
 pub open spec fn every_create_matching_pod_request_implies_at_after_create_pod_step(
-    vrs: VReplicaSetView, controller_id: int,
+    vrs: VReplicaSetView, cluster: Cluster, controller_id: int,
```

Comment:
It seems that the only thing you need from `cluster` is `installed_types`? If so I suggest only pass `installed_types` to this function, instead of the entire `cluster`.